python/TestHarness/testers/Tester.py

```diff
@@ -143,28 +114,24 @@ def getSuccessMessage(self):
 
     # Method to return status text (exodiff, crash, skipped because x, y and z etc)
     def getStatusMessage(self):
-        return self.status_message
+        return self.status.getStatusMessage()
 
-    # Method to return enumerated status (PASS, FAIL, SKIP, DIFF, SILENT, DELETED)
+    # Method to return status bucket tuple
     def getStatus(self):
-        if self.getStatusMessage() in self.status:
-            return self.status[self.getStatusMessage()]
-        return None
+        return self.status.getStatus()
 
     # Method to set the bucket status
     def setStatus(self, reason, bucket):
-        self.status_message = reason
-        self.status[reason] = bucket
+        self.status.setStatus(reason, bucket)
         return self.getStatus()
 
-    # Method to check if a test has failed. This method will return true iff a
+    # Method to check if a test has failed. This method will return true if a
```

Comment:
You may not realize this but this wasn't a typo:
https://en.wikipedia.org/wiki/If_and_only_if

It's fine, I don't think we need a pure mathematical definition here.